photon-core/src/main/java/org/photonvision/vision/camera/PVCameraInfo.java

```diff
@@ -189,7 +198,13 @@ public CameraType type() {
         public boolean equals(Object obj) {
             if (this == obj) return true;
             if (obj == null) return false;
-            return obj instanceof PVCameraInfo info && equals(info);
+            if (obj instanceof PVCSICameraInfo info) {
+                return baseName.equals(info.baseName) && equals(info);
+            }
+            if (obj instanceof PVCameraInfo info) {
+                return equals(info);
```

Comment:
why do we need this extra code? what does this equals normally end up checking for records?

Reply:
equals() in PVCameraInfo checks for uniquePath equality. each specific equals() method compares more properties to ensure equality. these checks were based off of what was on the camera matching code in the web ui code